spec/lib/s3_cors_rulesets_spec.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "s3_cors_rulesets"
+
+RSpec.describe S3CorsRulesets do
+  describe "#sync" do
+    let(:use_db_s3_config) { false }
+    let(:client) { Aws::S3::Client.new(stub_responses: true) }
+
+    it "does nothing when S3 is not set up" do
+      client.expects(:get_bucket_cors).never
+      sync_rules
+    end
+
+    context "when S3 is set up with global settings" do
+      let(:use_db_s3_config) { false }
+      before do
+        global_setting :s3_use_iam_profile, true
+        global_setting :s3_bucket, "s3-upload-bucket"
+        global_setting :s3_backup_bucket, "s3-backup-upload-bucket"
+        global_setting :s3_region, "us-west-2"
+      end
+
+      it "does nothing if !s3_install_cors_rule" do
+        SiteSetting.s3_install_cors_rule = false
+        sync_rules
+      end
```

Comment:
Minor but perhaps we can assert for the results returned from `sync_rules` here? Otherwise, it seems strange that there are no assertion in this test.

Reply:
Yes fair enough